clash-prelude/src/Clash/Explicit/Reset.hs

```diff
@@ -243,15 +244,14 @@ convertReset
   -> Clock domB
   -> Reset domA
   -> Reset domB
-convertReset clkA clkB (unsafeToHighPolarity -> rstA0) =
-  unsafeFromHighPolarity rstA2
+convertReset clkA clkB rstA0 = rstA2
  where
-  rstA1 = unsafeSynchronizer clkA clkB rstA0
+  rstA1 = unsafeSynchronizer clkA clkB $ unsafeToHighPolarity rstA0
   rstA2 =
-    case (resetKind @domA, resetKind @domB) of
-      (SSynchronous,  SSynchronous)  -> rstA1
-      (SAsynchronous, SAsynchronous) -> rstA1
-      (SSynchronous,  SAsynchronous) -> rstA1
-      (SAsynchronous, SSynchronous) ->
-        delay clkB enableGen True $
-          delay clkB enableGen True rstA1
+    case (sameDomain @domA @domB, resetKind @domA, resetKind @domB) of
+      (Just Refl, _, _)       -> rstA0
+      (_, _,  SAsynchronous)  -> unsafeFromHighPolarity rstA1
+      (_, _, _)               ->
+        unsafeFromHighPolarity
+          $ delay clkB enableGen True
+          $ delay clkB enableGen True rstA1
```

Comment:
These `delay`s are edge-sensitive, wouldn't it be neater if they would be level sensitive? If so, you could replace it with `resetSynchronizer clkB rstA1 enableGen` (+`unsafeFromHighPolarity` to `rstA1`).